### DISCUSSION TITLE
Add Test Results support for workflows

### DISCUSSION THREAD

**Discussion by dmattia** [community/community#12222] [Actions]
Our CI pipelines often run hundreds or thousands of tests. When a small number of tests fail, it is useful to developers to be able to quickly see the failing test names and outputs, if desired. We are migrating from a CI service that had such functionality by uploading JUNIT XML files containing the test output after testing jobs.

There are a number of marketplace actions that attempt to add this type of functionality to Actions, such as https://github.com/EnricoMi/publish-unit-test-result-action, but these all comment on the PRs, and are limited in their functionality. As an example, they cannot easily track historical test success rates to determine if a given test is flaky or not

**Comment by ethomson** [comment]
Thanks for the feedback, @dmattia - we'll take a look at adding this in the future.

**Comment by SupinePandora43** [comment]
My design idea:
![IMG_20220506_120749](https://user-images.githubusercontent.com/36124472/167083925-47a651fa-23f9-460c-8d31-a12660aee672.jpg)
